tempesta_fw/http.c

```diff
@@ -3353,6 +3353,20 @@ tfw_http_req_process(TfwConn *conn, const TfwFsmData *data)
 		TFW_INC_STAT_BH(clnt.msgs_otherr);
 	}
 
+	if (req->sess) {
```

Comment:
`req` must not be accessed after `tfw_http_send_resp()|tfw_cache_process()`. Request can be sent out there or even destroyed.

Saving `vhost` value should take place right after vhost is determined for the first time (same function, line 3219):
```diff
-	if (!req->vhost)
+	if (!req->vhost) {
		req->vhost = tfw_http_tbl_vhost((TfwMsg *)req, &block);
+		tfw_http_sess_pin_vhost(req);
+	}
	if (req->vhost)
		req->location = tfw_location_match(req->vhost,
						   &req->uri_path);
```

And I would like to move session-specific operations to `http_sess.c_

Reply:
Moved code to `http_sess.c`. Since `tfw_http_sess_pin_vhost()` is now part of API for sessions and can be used by some other code, it now also checks whenever `sess->vhost` was non-NULL to track references to `vhost` more accurately. Also removed outdated comments that I forgot to remove in the initial submission.